java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/resolution/MethodResolutionLogic.java

```diff
@@ -250,13 +250,20 @@ public int compare(MethodDeclaration m1, MethodDeclaration m2) {
                 if (m1 instanceof JavaParserMethodDeclaration && m2 instanceof JavaParserMethodDeclaration &&
                     ((JavaParserMethodDeclaration)m1).getWrappedNode().equals(((JavaParserMethodDeclaration)m2).getWrappedNode())) {
                     return 0;
-                } 
+                }
                 return 1;
             }
         });
         s.addAll(methods);
         List<MethodDeclaration> res = new ArrayList<>();
-        res.addAll(s);
+        Set<String> usedSignatures = new HashSet<>();
+        for (MethodDeclaration md : methods) {
+            String signature = md.getQualifiedSignature();
+            if (!usedSignatures.contains(signature)) {
+                usedSignatures.add(signature);
+                res.add(md);
+            }
+        }
```

Comment:
I think we can just use the new comparison logic you just added and remove the upper part of that method. I only wrote that upper part because at that time there was no getQualifiedSignature method for MethodDeclaration.
Or do you think the upper part is just a first phase to boost performance?